src/filter/QueenSacrificeFilter.ts

```diff
@@ -0,0 +1,106 @@
+import { VideoFilter } from "@/model/VideoFilter";
+import { Video } from "@/model/Video";
+import { Pgn } from "@/model/Pgn";
+import { usePgnsStore } from "@/stores/pgnsStore";
+
+/* 
+  There are some cases where a queen sacrifice may be wrongly detected:
+  - A blunder where the queen is captured without any compensation. However, it is unlikely that such games would be analyzed on the channel.
+  - A queen exchange combination, such as a check in-between move before recapturing the other queen.
+
+  There are also some cases where the queen sacrifice may not be detected:
+  - This function does not currently handle the scenario in which a queen exchange takes place and, later, a pawn is promoted to a queen, and then sacrificed.
+*/
+function isQueenSacrifice(pgn: string): boolean {
+  const moves = pgn
+    .replace(/\d+\.\s+/g, "") // Remove move numbers
+    .trim()
+    .split(/\s+/); // Split by spaces
+
+  const captureRegex = /([QRNB])?([a-h])?x([a-h][1-8])?/;
```

Comment:
This regex misses situation where two kinghts are located in single file. In that case notation is `N3xa5`

Reply:
Fixed to handle disambiguated moves.